### PR TITLE
Final Review

### PR DESCRIPTION
**Pull request by Fene-87**
In this project, I create a budget app that allows a user to keep track of his/her expenses.

### Accomplishments:
- Splash screen
    - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign up and log in pages
    - The user should be able to register in the app with full name, email and password (all mandatory).
    - The user can log into the app using email and password.
    - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
- Home page (categories page)
    - When the user logs in, they are presented with the categories page.
    - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
    - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
    - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- Transactions page
    - For a given category, the list of transactions is presented, ordered by the most recent.
    - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
    - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new category" page
    - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
    - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
   - When the user clicks on the "Back" button (<), the user navigates to the home page.
    - "Add a new transaction" page
- The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
    - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
    - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- Testing requirements
    - Create unit and integration tests for all the most important components of your RoR application.
- Deploy app on render

### CODE REVIEW

**Comment by addod19** [comment]
### Hi @Fene-87, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is invalid because you did not follow the correct Git-flow workflow, thus creating the PR from a descriptive feature branch like `budget-app` into the `dev` branch. At the moment, what we have now is GitHub-workflow

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Fene-87** [comment]
Hey @addod19 what do you mean? I dont understand you. I have created pull request from dev into main

**Comment by Fene-87** [comment]
The PR is from the dev branch @addod19. budget_app is the name of my application not a feature branch.

File: README.md [hunk 1]
```diff
@@ -1 +1,136 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+Click on the link to view the app.
+
+- [Live Demo](https://budget-tracker-lkkh.onrender.com)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (Omar-Muhamad) [inline README.md:49]: - [ ] When I try to test the deployed version of your app, an error is raised when I go to sign in or sign-up pages, Please fix that. As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=The%20project%20should%20be%20deployed%20and%20accessible%20online.) your app supposed to be functioning and accessible.

**Comment by addod19** [comment]
> Hey @addod19 what do you mean? I dont understand you. I have created pull request from dev into main

@Fene-87, yes I know and can see you have created the PR from dev into main, and that is called the `GitHub-flow`, what you are suppose to do is to follow the `Git-flow` which is creating the PR from a feature branch, you can name it whatever you want but it should be descriptive into the `dev` branch. I hope this is clearer

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Fene-87,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by addod19** [comment]
> The PR is from the dev branch @addod19. budget_app is the name of my application not a feature branch.

@Fene-87 , it is just an example I gave. You can ignore that name and use whatever name you please

File: README.md (hunk 1 above)
Comment (Fene-87) [reply README.md:49]: The live demo link has now been fixed.

**Review by Kamba56 — APPROVED** [review]
Hi @Fene-87, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
## Highlights
- [x] Good job using devise for authentication👏👏👏
- [x] Great job sticking to design !👏👏👏

## Optional suggestions

- Kindly add a demo video recorded presenting the features of the project.
The maximum length of the video should be 5 minutes.
The video contains a clear and objective description about the project in professional English. The topic and purpose of the project are made clear.
The video contains a demo of the project, highlighting all the project features.
The video contains a highlight about a piece of the code you find interesting or proud of making.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Fene-87** [comment]
Thank you so much for your time and feedback @Kamba56 I appreciate.

**Pull request merged**